### DISCUSSION TITLE
queues don't exists consumers

### DISCUSSION THREAD

**Discussion by MrRohwei** [rabbitmq/rabbitmq-website#1911] [General]
### Describe the bug

In my project, I have integrated rabbitmq using springboot. After deploying to the production environment, due to server upgrades in the production environment, there were no consumers under the queue, requiring me to restart my application service (During startup, it queries the queue names from the database via the CommandLineRunner#run method, dynamically creates queues using RabbitAdmin, and then adds them to the container via SimpleMessageListenerContainer#addQueues()). From what I understand, SimpleMessageListenerContainer should automatically manage the consumers for me, right?

### Reproduction steps

Upgraded the server during normal use

### Expected behavior

I hope the experts can help answer this.

**Comment by michaelklishin** [comment]
@MrRohwei that's a question for a Spring AMQP discussion forum. 

Spring AMQP enables [connection recovery](https://www.rabbitmq.com/client-libraries/java-api-guide#recovery) and even had its own version a bit before the Java client had it. 

But without an executable way to reproduce no one can tell you what the expected behavior should be.
